File: python/Queue.py
from DoublyLinkedList import DoublyLinkedList

class Queue:
    def __init__(self):
        self.list = DoublyLinkedList()
        
    def enqueue(self, data):
        self.list.insert_at(0, data)
        
    def dequeue(self):
        try:
            return self.list.delete_last()
        except IndexError:
            return None
    
    def front(self):
        return self.list.tail
    
    def is_empty(self):
        return self.list.size == 0

File: js/Queue.mjs
import { DoublyLinkedList } from './DoublyLinkedList.mjs';

class Queue {
    constructor() {
        this.list = new DoublyLinkedList();
    }
    
    enqueue(data) {
        this.list.insert(0, data);
    }

    dequeue() {
        try {
            return this.list.deleteLast();
        } catch (error) {
            return null;
        }
    }
    
    front() {
        return this.list.tail;
    }

    isEmpty() {
        return (this.list.count === 0);
    }
    
}

export { Queue }